### ISSUE TITLE
Constructor Function: Parameter Name, Zero-Arity

### ISSUE THREAD

**Issue by ChristianGruen** [qt4cg/qtspecs#658]
The parameter name for constructor functions is `$arg`: https://qt4cg.org/specifications/xpath-functions-40/Overview.html#constructor-functions

We should change it to `$value`, in alignment with the XQFO functions:

```xquery
fn:string(value := 123),
xs:string(value := 123)
```

**Comment by michaelhkay** [comment]
See also #617.

While on the topic, for further alignment, should we allow zero-arity constructor functions with the argument defaulting to the context item?

**Comment by ChristianGruen** [comment]
Absolutely, yes!

**Comment by michaelhkay** [comment]
Note, a bug has crept into the spec. In 3.1 the signature is given (in each case) as:

xs:string($arg as xs:anyAtomicType?) as xs:string?

but in 4.0 we have lost the question mark:

```
xs:string(
   $arg as xs:anyAtomicType
) as xs:string?
```
This seems to be a markup/stylesheet problem. In the source XML the argument is given as

```
 type="xs:anyAtomicType" emptyOk="yes"
```

And we seem to have dropped support for the `emptyOk="yes"` formulation.

**Comment by michaelhkay** [comment]
Closed by application of PR #662
